src/puya/awst_build/eb/arc4/arrays.py

```diff
@@ -235,6 +240,46 @@ def __init__(self, location: SourceLocation):
             element_wtype, array_size=array_size, alias="address"
         )
 
+    def call(
+        self,
+        args: Sequence[ExpressionBuilder | Literal],
+        arg_kinds: list[mypy.nodes.ArgKind],
+        arg_names: list[str | None],
+        location: SourceLocation,
+        original_expr: mypy.nodes.CallExpr,
+    ) -> ExpressionBuilder:
+        match args:
+            case (ExpressionBuilder() as eb,) if eb.rvalue().wtype == wtypes.account_wtype:
+                address_bytes: Expression = get_bytes_expr(eb.rvalue())
+            case (eb_or_literal,):
```

Comment:
Could this handle Literal separately so the length check can be done at compile time in that case?

Reply:
The optimiser should pick up the const value and optimise it away